### PR TITLE
Add card wallet, and ideal, card_present PaymentMethod Types

### PR DESCRIPTION
**Pull request by yuki-stripe**
## Summary
* Add STPPaymentMethodCardPresent & STPPaymentMethodTypeCardPresent
* Add STPPaymentMethodIdeal & STPPaymentMethodTypeIdeal
* Add STPPaymentMethodCardWallet, STPPaymentMethodCardWalletMasterpass, STPPaymentMethodCardWalletVisaCheckout

## Motivation
Support for PaymentMethod

## Testing
STPPaymentMethodCardWalletTest
STPPaymentMethodCardWalletMasterpassTest
STPPaymentMethodCardWalletVisaCheckoutTest
STPPaymentMethodIdealTest
Updated STPPaymentMethodTest for new types

### CODE REVIEW

**Comment by csabol-stripe** [comment]
I think the base branch for this is set incorrectly?

**Comment by yuki-stripe** [comment]
The base branch was set correctly so I'm not sure what happened, but the diff looks correct now. ptal @csabol-stripe

File: Stripe/PublicHeaders/STPPaymentMethod.h [hunk 1]
```diff
@@ -53,6 +53,16 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, nullable, readonly) STPPaymentMethodCard *card;
 
+/**
+ If this is a iDEAL PaymentMethod (ie `self.type == STPPaymentMethodTypeIdeal`), this contains additional details.
+ */
+@property (nonatomic, nullable, readonly) STPPaymentMethodIdeal *ideal;
```
Comment (csabol-stripe) [inline Stripe/PublicHeaders/STPPaymentMethod.h]: `iDEAL`

File: Stripe/PublicHeaders/STPPaymentMethodCardWallet.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#import <Foundation/Foundation.h>
+
+#import "STPAPIResponseDecodable.h"
+
+@class STPPaymentMethodCardWalletVisaCheckout, STPPaymentMethodCardWalletMasterpass;
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, STPPaymentMethodCardWalletType) {
+    STPPaymentMethodCardWalletTypeAmexExpressCheckout,
```
Comment (csabol-stripe) [inline Stripe/PublicHeaders/STPPaymentMethodCardWallet.h]: I think these will need doc-comments or jazzy will complain

File: Stripe/PublicHeaders/STPPaymentMethodEnums.h [hunk 3]
```diff
@@ -15,6 +15,16 @@ typedef NS_ENUM(NSUInteger, STPPaymentMethodType) {
      */
     STPPaymentMethodTypeCard,
     
+    /**
+     An iDEAL payment method.
+     */
+    STPPaymentMethodTypeIdeal,
```
Comment (csabol-stripe) [inline Stripe/PublicHeaders/STPPaymentMethodEnums.h]: `STPPaymentMethodTypeiDEAL`

File: Stripe/PublicHeaders/STPPaymentMethodIdeal.h [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+//  Copyright © 2019 Stripe, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "STPAPIResponseDecodable.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface STPPaymentMethodIdeal : NSObject <STPAPIResponseDecodable>
```
Comment (csabol-stripe) [inline Stripe/PublicHeaders/STPPaymentMethodIdeal.h]: `STPPaymentMethodiDEAL` or maybe `STPPaymentMethodIDEAL`?

File: Stripe/PublicHeaders/STPPaymentMethodIdeal.h [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ The customer’s bank.
+ */
+@property (nonatomic, nullable, readonly) NSString *bank;
```
Comment (csabol-stripe) [inline Stripe/PublicHeaders/STPPaymentMethodIdeal.h]: Is this the name of the bank? If so can we rename to reflect (or so it's more descriptive of whatever it is) to something like `bankName`?

File: Stripe/PublicHeaders/STPPaymentMethodIdeal.h [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ The Bank Identifier Code of the customer’s bank.
+ */
+@property (nonatomic, nullable, readonly) NSString *bic;
```
Comment (csabol-stripe) [inline Stripe/PublicHeaders/STPPaymentMethodIdeal.h]: Any opposition to just naming this `bankIdentifierCode`?

File: Stripe/PublicHeaders/STPPaymentMethodIdealParams.h [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+#import "STPFormEncodable.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface STPPaymentMethodIdealParams : NSObject <STPFormEncodable>
+
+/**
+ The customer’s bank.
+ */
+@property (nonatomic, nullable, copy) NSString *bank;
```
Comment (csabol-stripe) [inline Stripe/PublicHeaders/STPPaymentMethodIdealParams.h]: same as above

File: Stripe/PublicHeaders/STPPaymentMethodParams.h [hunk 8]
```diff
@@ -48,6 +48,11 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, strong, nullable) STPPaymentMethodCardParams *card;
 
+/**
+ If this is a iDEAL PaymentMethod, this contains details about user's bank.
+ */
+@property (nonatomic, nullable) STPPaymentMethodIdealParams *ideal;
```
Comment (csabol-stripe) [inline Stripe/PublicHeaders/STPPaymentMethodParams.h]: `iDEAL`

File: Stripe/PublicHeaders/STPPaymentMethodParams.h [hunk 9]
```diff
@@ -64,6 +69,17 @@ NS_ASSUME_NONNULL_BEGIN
...
                                       metadata:(nullable NSDictionary<NSString *, NSString *> *)metadata;
 
+/**
+ Creates params for a card PaymentMethod.
+ 
+ @param ideal                An object containing the user's iDEAL bank details.
+ @param billingDetails      An object containing the user's billing details.
+ @param metadata            Additional information to attach to the PaymentMethod.
+ */
++ (STPPaymentMethodParams *)paramsWithIdeal:(STPPaymentMethodIdealParams *)ideal
```
Comment (csabol-stripe) [inline Stripe/PublicHeaders/STPPaymentMethodParams.h]: `iDEAL`

File: Stripe/STPPaymentMethod.m [hunk 10]
```diff
@@ -20,6 +22,8 @@ @interface STPPaymentMethod ()
 @property (nonatomic, readwrite) STPPaymentMethodType type;
 @property (nonatomic, strong, nullable, readwrite) STPPaymentMethodBillingDetails *billingDetails;
 @property (nonatomic, strong, nullable, readwrite) STPPaymentMethodCard *card;
+@property (nonatomic, strong, nullable, readwrite) STPPaymentMethodIdeal *ideal;
```
Comment (csabol-stripe) [inline Stripe/STPPaymentMethod.m]: `iDEAL`

File: Stripe/STPPaymentMethodCardWallet+Private.h [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+//
+//  Created by Yuki Tokuhiro on 3/9/19.
+//  Copyright © 2019 Stripe, Inc. All rights reserved.
+//
+
+#import "STPPaymentMethodCardWallet.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface STPPaymentMethodCardWallet ()
```
Comment (csabol-stripe) [inline Stripe/STPPaymentMethodCardWallet+Private.h]: name interface

**Pull request merged**